src/libcrun/container.c

```diff
@@ -3706,6 +3707,131 @@ libcrun_container_update_from_values (libcrun_context_t *context, const char *id
   return ret;
 }
 
+void populate_array_field(const char*** field, const char* array[], size_t numElements){
+    *field = malloc(numElements * sizeof(char *));
+    size_t i;
+    for (i = 0; i < numElements; i++) {
+        (*field)[i] = strdup(array[i]);
+    }
+
+    (*field)[i] = NULL;  // Terminate the array with NULL
+}
+
+int libcrun_container_print_features(struct features_info_s *info)
+{
+    // Hardcoded feature information
+    info->ociVersionMin = "1.0.0";
+    info->ociVersionMax = "1.0.2-dev";
+
+    // Hardcoded hooks
+    const char *hooks[] = {
+        "prestart",
+        "createRuntime",
+        "createContainer",
+        "startContainer",
+        "poststart",
+        "poststop"
+    };
+
+    // Hardcode the values for namespaces
+    static const char *namespaces[] = {
+        "cgroup",
+        "ipc",
+        "mount",
+        "network",
+        "pid",
+        "user",
+        "uts"
+    };
+
+   // Hardcode the values for capabilities
+    static const char *capabilities[] = {
+        "CAP_CHOWN",
+        "CAP_DAC_OVERRIDE",
+        "CAP_DAC_READ_SEARCH",
+        "CAP_FOWNER",
+        "CAP_FSETID",
+        "CAP_KILL",
+        "CAP_SETGID",
+        "CAP_SETUID",
+        "CAP_SETPCAP",
+        "CAP_LINUX_IMMUTABLE",
+        "CAP_NET_BIND_SERVICE",
+        "CAP_NET_BROADCAST",
+        "CAP_NET_ADMIN",
+        "CAP_NET_RAW",
+        "CAP_IPC_LOCK",
+        "CAP_IPC_OWNER",
+        "CAP_SYS_MODULE",
+        "CAP_SYS_RAWIO",
+        "CAP_SYS_CHROOT",
+        "CAP_SYS_PTRACE",
+        "CAP_SYS_PACCT",
+        "CAP_SYS_ADMIN",
+        "CAP_SYS_BOOT",
+        "CAP_SYS_NICE",
+        "CAP_SYS_RESOURCE",
+        "CAP_SYS_TIME",
+        "CAP_SYS_TTY_CONFIG",
+        "CAP_MKNOD",
+        "CAP_LEASE",
+        "CAP_AUDIT_WRITE",
+        "CAP_AUDIT_CONTROL",
+        "CAP_SETFCAP",
+        "CAP_MAC_OVERRIDE",
+        "CAP_MAC_ADMIN",
+        "CAP_SYSLOG",
+        "CAP_WAKE_ALARM",
+        "CAP_BLOCK_SUSPEND",
+        "CAP_AUDIT_READ",
+        "CAP_PERFMON",
+        "CAP_BPF",
+        "CAP_CHECKPOINT_RESTORE"
+    };
+
+    // Populate hooks
+    size_t numHooks = sizeof(hooks) / sizeof(hooks[0]);
+    populate_array_field(&(info->hooks), hooks, numHooks);
+
+    // Retrieve mount options from wordlist
+    size_t numMountOptions;
+    const struct propagation_flags_s* mountOptionsList = get_mount_flags_from_wordlist();
+
+    // Calculate the number of mount options
+    size_t count = 0;
+    while (mountOptionsList[count].name != NULL) {
+        count++;
+    }
```

Comment:
we are assuming it is NULL terminated here, so we must allocate an extra element at the end of the array